node/bft/src/primary.rs

```diff
@@ -955,14 +955,28 @@ impl<N: Network> Primary<N> {
             bail!("Received a batch certificate for myself ({author})");
         }
 
-        // Store the certificate, after ensuring it is valid.
+        // Ensure that the incoming certificate is valid.
+        self.storage.check_incoming_certificate(&certificate)?;
```

Comment:
If the certificate is more than 100 rounds in the future, would we just fail here? Could that happen with a correct/honest peer if there is a network issue?

Reply:
Yes, we would fail because the validator would be unable to calculate the active committee for the certificate's round, given that its blockchain is too behind compared to the round. This ties in with more general liveness issues, which need further study.